blas/impl/KokkosBlas2_ger_spec.hpp

```diff
@@ -0,0 +1,214 @@
+                                         >                                         \
+                           , Kokkos::View< const SCALAR*                           \
+                                         , LAYOUT                                  \
+                                         , Kokkos::Device<EXEC_SPACE, MEM_SPACE>   \
+                                         , Kokkos::MemoryTraits<Kokkos::Unmanaged> \
+                                         >                                         \
+                           , Kokkos::View< SCALAR**                                \
+                                         , LAYOUT                                  \
+                                         , Kokkos::Device<EXEC_SPACE, MEM_SPACE>   \
+                                         , Kokkos::MemoryTraits<Kokkos::Unmanaged> \
+                                         >                                         \
+                           > {                                                     \
+    enum : bool { value = true };                                                  \
+  };
+
+// Include the actual specialization declarations
+#include <KokkosBlas2_ger_tpl_spec_avail.hpp>
+#include <generated_specializations_hpp/KokkosBlas2_ger_eti_spec_avail.hpp>
+
+namespace KokkosBlas {
+namespace Impl {
+
+//
+// ger
+//
+
+// Implementation of KokkosBlas::ger.
+template < class XViewType
+         , class YViewType
+         , class AViewType
+         , bool tpl_spec_avail = ger_tpl_spec_avail<XViewType, YViewType, AViewType>::value
+         , bool eti_spec_avail = ger_eti_spec_avail<XViewType, YViewType, AViewType>::value
+         >
+struct GER {
+  static void ger( const typename AViewType::execution_space  & space
+                 , const          char                          trans[]
+                 , const typename AViewType::const_value_type & alpha
+                 , const          XViewType                   & x
+                 , const          YViewType                   & y
+                 , const          AViewType                   & A
+                 )
+#if !defined(KOKKOSKERNELS_ETI_ONLY) || KOKKOSKERNELS_IMPL_COMPILE_LIBRARY
+  {
+    KOKKOS_IMPL_DO_NOT_USE_PRINTF( "Entering KokkosBlas::Impl::Ger::ger()\n" );
+
+    static_assert(Kokkos::is_view<XViewType>::value, "XViewType must be a Kokkos::View.");
+    static_assert(Kokkos::is_view<YViewType>::value, "YViewType must be a Kokkos::View.");
+    static_assert(Kokkos::is_view<AViewType>::value, "AViewType must be a Kokkos::View.");
+
+    static_assert(static_cast<int>(XViewType::rank) == 1, "XViewType must have rank 1.");
+    static_assert(static_cast<int>(YViewType::rank) == 1, "YViewType must have rank 1.");
+    static_assert(static_cast<int>(AViewType::rank) == 2, "AViewType must have rank 2.");
+
+    if ((trans[0] == 'T') ||
+        (trans[0] == 't') ||
+        (trans[0] == 'H') ||
+        (trans[0] == 'h')) {
+      // Ok
```

Comment:
This can probably go

Reply:
Agreed. Put only one check at blas/src/KokkosBlas2_ger.hpp.